### DISCUSSION TITLE
is it good to use many useEffect in a component ? is there any better perfomant way I can write the same?

### DISCUSSION THREAD

**Discussion by jaimeenkoffeekodes** [vercel/next.js#66163] [Help]
![manyUseEffect](https://github.com/vercel/next.js/assets/149799052/cd50bed5-fb61-4f19-941f-244e94b3049d)

**Comment by icyJoseph** [comment]
Hi,

In a way, many useState, and many useEffect are preferred, as long as theres some separation of concerns between them.

In your case, I'd say it is not a React problem, but a JavaScript one.

You are allocation these arrays many times, and running `.includes`, which walks the array to find the image you want to check.

I'd say, if it makes business sense, move those array literals, to outside the function, give them names, and use them as constants throughout your component.

And then, wrap them in `new Set(MY_CONSTANT_ARRAY)` why? Well, a set is more performant to do look ups! https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/has
